File: orders-service/src/main/java/org/example/ordersservice/models/Mensaje.java
package org.example.ordersservice.models;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "mensaje")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Mensaje {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDateTime fecha;

    @Column(nullable = false, length = 1000)
    private String mensaje;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "emisor_id")
    private User emisor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "receptor_id")
    private User receptor;

    private boolean leido;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof User)) return false;
        User user = (User) o;
        return id != null && id.equals(user.getId());
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }


}

File: front/src/app/types/mensaje.ts
export interface Mensaje {
    id?: number;
    fecha: string;
    mensaje: string;
    emisorId?: number;
    receptorId?: number;
    leido: boolean;
}

export interface MensajeInputDto {
    contenido: string;
    receptorId: number;
}

export interface MensajeOutputDto {
    id: number;
    contenido: string;
    fechaEnvio: string;
    emisorId: number;
    nombreEmisor: string;
    receptorId: number;
    nombreReceptor: string;
}